lightning/Cargo.toml

```diff
@@ -26,25 +26,31 @@ max_level_debug = []
 unsafe_revoked_tx_signing = []
 unstable = []
 
-no_std = ["hashbrown", "bitcoin/no-std"]
-std = ["bitcoin/std"]
+no_std = ["hashbrown", "bitcoin/no-std", "core2/alloc", "bitcoin/no-std"]
+std = ["bitcoin/std", "bitcoin/std"]
 
 default = ["std"]
 
 [dependencies]
-bitcoin = "0.27"
+bitcoin = { version = "0.27", default-features = false, features = ["secp-recovery"] }
+# TODO maybe bitcoin no-std should pull in this feature?
+secp256k1 = { version = "0.20.2", default-features = false, features = ["alloc"] }
 
 hashbrown = { version = "0.11", optional = true }
 hex = { version = "0.3", optional = true }
 regex = { version = "0.1.80", optional = true }
 
+core2 = { version = "0.3.0", optional = true, default-features = false }
+
 [dev-dependencies]
 hex = "0.3"
 regex = "0.1.80"
+secp256k1 = { version = "0.20.2", default-features = false, features = ["alloc"] }
 
 [dev-dependencies.bitcoin]
 version = "0.27"
-features = ["bitcoinconsensus"]
+default-features = false
```

Comment:
Are you saying that libbitcoinconsensus is dependent on `std` anyway, so there's no point in turning it off?  shouldn't that library be made `no_std`?

Reply:
Hmm, yea, I guess it could be made no_std eventually, though I doubt anyone is gonna sit down and fight to get it there.

Reply:
So if I remove `default-features = false` then `cargo test --no-default-features --features no_std` fails.  I think that's because the `bitcoin` crate has the `std` feature on, but this crate doesn't, so there's a mismatch in the `bitcoin` API signatures.